tasks/winbuildscripts/Generate-Pyzip.ps1

```diff
@@ -10,49 +10,74 @@ param (
 # https://www.python.org/ftp/python/3.8.0/python-3.8.0-amd64.exe
 # https://www.python.org/ftp/python/3.8.0/python-3.8.0.exe
 
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+
 $maj, $min, $patch = $Version.Split(".")
 Write-Host -ForegroundColor Green "Installing Python version $maj $min $patch"
 
-if($maj -eq "2") {
-    $url = "https://www.python.org/ftp/python/$Version/python-$Version.amd64.msi"
-} elseif ($maj -eq "3"){
-    $url = "https://www.python.org/ftp/python/$Version/python-$Version-amd64.exe"
-} else {
-    Write-Host -ForegroundColor Red "Unknown major version $Maj.  I don't know how to do this"
-    exit 1
-}
-$outzip = "python-windows-$Version-amd64.zip"
-if($x86 -eq $true){
-    if($maj -eq "2") {
-        $url = "https://www.python.org/ftp/python/$Version/python-$Version.msi"
+function Get-Installer-Url {
+    param (
+        [string]$version
+    )
+
+    $maj, $min, $patch = $Version.Split(".")
+    if ($maj -eq "2") {
+        $url = "https://www.python.org/ftp/python/$Version/python-$Version.amd64.msi"
     } elseif ($maj -eq "3"){
-        $url = "https://www.python.org/ftp/python/$Version/python-$Version.exe"
+        $url = "https://www.python.org/ftp/python/$Version/python-$Version-amd64.exe"
+    } else {
+        Write-Host -ForegroundColor Red "Unknown major version $Maj.  I don't know how to do this"
+        exit 1
     }
+    
+    if ($x86 -eq $true) {
+        if ($maj -eq "2") {
+            $url = "https://www.python.org/ftp/python/$Version/python-$Version.msi"
+        } elseif ($maj -eq "3"){
+            $url = "https://www.python.org/ftp/python/$Version/python-$Version.exe"
+        }
+    }
+    return $url
+}
+
+$outzip = "python-windows-$Version-amd64.zip"
+if ($x86 -eq $true) {
     $outzip = "python-windows-$Version-x86.zip"
 }
-[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+$url = Get-Installer-Url -version $version
 $dlfilename = ($url -split '/')[-1]
 $dlfullpath = "$Env:TEMP\$dlfilename"
-Write-Host -ForegroundColor Green "Downloading file from $url..."
-(New-Object System.Net.WebClient).DownloadFile($url, $dlfullpath)
-Write-Host -ForegroundColor Green "... Done."
 
+function Get-Installer {
+    param  (
+        [string]$url,
+        [string]$dlfullpath
+    )
 
-## do md5 check if we have it
-if($md5sum){
-    $downloadedhash = (Get-FileHash -Algorithm md5 $dlfullpath).Hash.ToLower()
-    if($downloadedhash -ne $md5sum.ToLower()){
-        Write-Host -ForegroundColor Red "MD5Sums Don't Match"
-        exit -1
+    Write-Host -ForegroundColor Green "Downloading file from $url..."
+    (New-Object System.Net.WebClient).DownloadFile($url, $dlfullpath)
+    Write-Host -ForegroundColor Green "... Done."
+
+    ## do md5 check if we have it
+    if($md5sum){
```

Comment:
I know this will work (the outer param is global scope), but probably want to make md5sum a param to this function for clarity and pass from the outer.